src/txn/txn.c

```diff
@@ -1663,7 +1663,7 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
         if (op->type == WT_TXN_OP_REF_DELETE) {
             WT_REF_LOCK(session, op->u.ref, &previous_state);
             if (previous_state == WT_REF_DELETED)
-                op->u.ref->ft_info.del->committed = 1;
+                op->u.ref->ft_info.del->resolved = true;
```

Comment:
I don't understand why if the previous state is WT_REF_DELETED, the fast truncate is resolved.

Reply:
If it is rolledback, it is not resolved?

Reply:
> I don't understand why if the previous state is WT_REF_DELETED, the fast truncate is resolved.

If the WT_REF.type hasn't changed from WT_REF_DELETED, we started a fast-truncate and nothing changed, and we committed it, so we mark the WT_REF.del as "resolved".

If the WT_REF.type changed, it means we instantiated the page after the fast-truncate started, and nothing in the WT_PAGE_DELETED structure is useful after commit, so we discard that structure. Because WT_REF.ft_info.del is a union with WT_REF.ft_info.update, freeing the update field frees everything.

The comment above that code has some text, but it may need enhancing.

Reply:
> If it is rolledback, it is not resolved?

__wt_txn_rollback() calls __wt_delete_page_rollback() which has some similar work:
```
    /*
     * If the page is still "deleted", it's as we left it, simply reset the state. Otherwise, the
     * page is in an in-memory state, which means it was instantiated at some point. Walk any list
     * of update structures and abort them. We can't use the normal read path to get the pages with
     * updates (the original page may have split, so there many be more than one page), because the
     * session may have closed the cursor, we no longer have the reference to the tree required for
     * a hazard pointer. We're safe since pages with unresolved transactions aren't going anywhere.
     */
    if (current_state == WT_REF_DELETED)
        current_state = ref->ft_info.del->previous_ref_state;
    else if ((updp = ref->ft_info.update) != NULL)
        for (; *updp != NULL; ++updp)
            (*updp)->txnid = WT_TXN_ABORTED;

    /*
     * Don't set the WT_PAGE_DELETED transaction ID to aborted, discard any WT_UPDATE list or set
     * the resolved flag to 1; instead, discard the structures, it has the same effect. It's a
     * single call, they're a union of two pointers.
     */
    __wt_free(session, ref->ft_info.del);
```

Reply:
I see. The name resolved is a bit confusing may be change it back to committed?

Reply:
Ummmm, sure... :)

Reply:
Fixed in 664f38e.